storage/store.go

```diff
@@ -1293,6 +1344,11 @@ func (s *Store) ExecuteCmd(ctx context.Context, args proto.Request) (proto.Respo
 		// advances the local node's clock to a high water mark from
 		// amongst all nodes with which it has interacted.
 		s.ctx.Clock.Update(header.Timestamp)
+	} else {
+		// If the request has a zero timestamp, initialize to this node's clock.
+		// TODO(tschottdorf): the above comment was there, but not the line below.
+		// Check if this is done anywhere else.
+		header.Timestamp = s.ctx.Clock.Now()
```

Comment:
a comment was there. I added the TODO to figure out whether it got lost or whether the comment was stale.